src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -770,7 +792,7 @@ export class FolderMatch extends Disposable {
 
 		const removed = [];
 		for (const match of fileMatches as FileMatch[]) {
-			if (this.fileMatches().includes(match)) {
+			if (this.getFileMatch(match.resource)) {
```

Comment:
Can't this just access `this._fileMatches` directly? If not, looks like getFileMatch should be private at least